core/backoff/retry.go

```diff
@@ -70,66 +71,89 @@ type Strategy struct {
 //	    MaxRetries: 5,
 //	    Delay: 100,
 //	    Exponential: true,
-//	    MaxDuration: 10 * time.Second,
+//	    MaxWait: 10 * time.Second,
 //	})
 //	if err != nil {
 //	    fmt.Println("Failed to connect to database after retries:", err)
 //	}
 func Retry(scope string, f func() error, s Strategy) error {
+	cid := crypto.Id()
```

Comment:
better to pass the id (and scope) through some context; but that can be done as a separate PR.